Un-ignore DeprecationWarning

uranusjr opened the pull request:
I introduced a flag in #20217 to help us not accidentally use deprecated context variables. This works locally, but I failed to notice it does not work in CI because our scripts explicitly pass `--pythonwarnings=ignore::DeprecationWarning`, which overrides my added config. Consequently, quite several deprecated usages leaked into `main`.

This PR properly fixes all those deprecated usages. Also, since no explainations were offered at the time the flag was added (it’s been there ever since Pytest was adopted as the test runner), I removed that ignore flag entirely. This should help us identify more deprecated usages (there are quite several) and fix them eventually to work toward #19788.

uranusjr committed:
Un-ignore DeprecationWarning

uranusjr committed:
Fix deprecation warnings in external task sensors

Unfortunately, external task sensors reference "execution date" quite
extensively, and even include it as a part of its public API (both that
exact variable, and things like execution_offset and excution_date_fn).
It is definitely possible to migrate them to something else, but it is
some substential work, and I actually suspect these aren't used widely,
if by anyone at all, since we did not receive a single report pointing
out those are currently emitting DeprecationWarning messages from core
Airflow code. So I'm going to leave the public interface alone for now,
and only change the internals to not emit warnings.

uranusjr committed:
Fix deprecation warnings from DayOfWeekSensor

Similar to ExternalTaskSensor, the execution date terminology is
unfortunately a part of its public API. This one is easier to deprecate,
but again, it's entirely unclear how many people are actually using
this, especially now we have async triggers. So I'm leaving the
interface alone.

uranusjr committed:
Fix deprecation warning in BranchDateTimeOperator

Again, execution date is unfortunately a part of the operator's public
interface. But again, better to leave it again. :shrug:

uranusjr committed:
Don't use execution_date in EmailOperator test

uranusjr committed:
Use logical date directly in ShortCircuitOperator

uranusjr committed:
Allow deprecations in some Python callable tests

Where we actually want to make sure those arguments work properly.

uranusjr committed:
No execution_date in BranchDayOfWeekOperator test

uranusjr committed:
No execution_date in HttpSensor test

uranusjr committed:
Do not eagerly resolve context in HTTP operators

To avoid deprecated context variables emitting warnings, we must call
determine_kwargs directly on the context mapping first, instead of using
the make_kwargs_callable wrapper, which uses **kwargs and eagerly access
all the members.

uranusjr committed:
Use our own Jinja renderer for log filename

Similar to other Jinja rendering situations, we can't just pass
**context into Template.render().

uranusjr committed:
Remove execution_date from BranchDayOfWeekOperator

Similar to DayOfWeekSensor and BranchDateTimeOperator, this also has
execution date in its public interface. But let's not deal with it now.

uranusjr committed:
Use custom Jinja renderer for custom log formatter

Same as the regular handler.

uranusjr committed:
Replace eager Jinja rendering on email-sending

More of the same thing.

uranusjr committed:
Remove **kwargs and mark deprecated calls in tests

uranusjr committed:
Allow deprecated in print_the_context tests

ashb commented on airflow/utils/context.py:
```diff
@@ -191,7 +191,14 @@ def items(self):
+
+    This is implemented as a free function because the ``Context`` type is
+    "faked" as a ``TypedDict`` in ``context.pyi``, which cannot have custom
+    functions.
```
```suggestion
    functions.
    
    :meta private:
```

as we probably don't want this to be part of the "public API".

ashb reviewed (commented):
Looking good so far.

uranusjr committed:
Refactor to use lazy proxy on callable with **kw

Due to how Python works, we can't make the keyword argument dict lazy,
and calling fn(**context) would emit deprecation warnings too eagerly.
The previous warning strategy with lazy-object-proxy is brought back for
this particular use case, and only deprecated entries are converted to
lazy proxies if the context is being unpacked, to keep possible
incompatibilities minimal.

uranusjr committed:
Remove context warning ignores no longer needed

uranusjr committed:
Remove Context-related functions from docs

uranusjr committed:
Remove option to ignore deprecation warnings on CI

ashb reviewed (approved):
Have we removed the warning filter from the ci cmdline?
